front-end/src/AdminBar/AdminView.js

```diff
@@ -1,91 +1,84 @@
 import React from "react";
 import { addOrRemove, clear } from "../utils/api";
-import { AdminPanel, FocusInput, ColoredButton } from "./adminStyles";
-import {
-  ESCAPE_KEY,
-  ENTER_KEY,
-  NUMPAD_ENTER_KEY,
-  BACKSPACE,
-  DELETE
-} from "../utils/keypadNumbers";
+import { AdminForm, FocusInput, ColoredButton } from "./adminStyles";
+
+const MAX_NUMBER_LENGTH = 8;
+const filterNumeric = str => str.replace(/\D/g,'');
 
 class AdminView extends React.Component {
   constructor(props) {
     super(props);
     this.state = { typed: "" };
+    this.inputRef = React.createRef();
+    this.onFormSubmit = this.onFormSubmit.bind(this);
```

Comment:
Any reason onFormSubmit needs to be bound instead of being an arrow function? :)

Reply:
Oh, not at all! Just out of habit, haven't really been using "arrow methods" (or whatever they're called.. :joy:) much yet :)